sale_order_type/models/sale.py

```diff
@@ -125,6 +125,13 @@ def _prepare_invoice(self):
             res["sale_type_id"] = self.type_id.id
         return res
 
+    def copy(self, default=None):
```

Comment:
Doing this in the method `copy_data` you avoid an extra write, although you would need to do a browse over ` type_id` ID result if exists.